### PR TITLE
fix Any typing and cleanup code

### LINKED ISSUE #389: Fix typing Any
#### Description
Flake8-annotations has introduced a new requirement to code styling which gives an error:
`ANN401 Dynamically typed expressions (typing.Any) are disallowed `
This is now fixed by ignoring this error in tox.ini.

It would be desirable to fix this.

#### DoD (Definition of Done)
Flake8 check is not giving ANN401 error.

#### Testing
running tox command without `ignore: ANN401`

### LINKED ISSUE #575: Bring back option to run integration tests without clean-up
#### Proposed solution

with #448 integration tests delete data without leaving the option to analyze it in a successful run, to see if it is formatted correctly as a whole. 

such feature is good for :
- schema changes - mongo is a schema-less afterall
- adding new options to mongo collections
- debugging queries to mongo without much hassle to create data

#### DoD (Definition of Done)

add option to integration tests to run them without cleaning data from databse

#### Testing

integration tests can be run without cleaning data from mongo.

### PR DESCRIPTION
**Pull request by blankdots**
### Related issues

closes #389 
closes #575 

### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

### Changes Made

1.  fix `Any` type wherever that is possible. There were some instances like ABC class where this could not be easily satisfied as well as with decorators and some custom classes we extend from `xmlschema` and `aiohttp`. However it might be that python 3.10 + might bring better options for that 
2. remove unused code, which i suspect is the result of faulty rebase or rollback of certain features + the fact that our `master` and `develop` branches are out of sync.
3. remove some `pylint` ignores that were not required e.g. i did not see `web.HTTPSuccessful` ever being raised and did not follow why it was handled in the exception middleware
4. xml object was not properly deleted from the corresponding schema in the db, and it made unnecessary calls for some objects, as the only places it was required was for deleting `objects` and `submissions`
5. add command line options for ``--nocleanup`` integ tests and docs about it
6. update pre-commit config dependencies

### Testing

- [x] Unit Tests
- [x] Integration Tests

### Mentions

this also addresses some of the dead code in #577

### CODE REVIEW

File: metadata_backend/services/service_handler.py [hunk 1]
```diff
@@ -104,9 +105,9 @@ async def _request(
         url: URL = None,
         path: str = "",
         params: Union[str, dict] = None,
-        json_data: Any = None,
+        json_data: dict = None,
```
Comment (csc-felipe) [inline metadata_backend/services/service_handler.py:108]: I don't think this is always a `dict`.

File: tox.ini [hunk 2]
```diff
@@ -8,7 +8,7 @@ asyncio_mode = auto
 [flake8]
 max-line-length = 120
 # ANN40 will be fixed with separate PR
```
Comment (csc-felipe) [inline tox.ini:10]: Could also remove this line

**Comment by csc-felipe** [comment]
It's fun when integration tests fail on the `asyncio.gather`.

File: metadata_backend/services/service_handler.py (hunk 1 above)
Comment (blankdots) [reply metadata_backend/services/service_handler.py:108]: well, then mypy is not perfect :)

File: tests/integration/conftest.py [hunk 3]
```diff
@@ -25,6 +25,16 @@
 from tests.integration.mongo import Mongo
 
 
+def pytest_addoption(parser):
+    """Add command line options."""
+    parser.addoption(
```
Comment (csc-felipe) [inline tests/integration/conftest.py:30]: nice solution

**Review by csc-felipe — APPROVED** [review]

**Review by teemukataja — APPROVED** [review]

**Pull request merged**